packages/mcp-auth/src/routers/create-delegated-router.ts

```diff
@@ -26,19 +30,35 @@ import { serverMetadataPaths } from '../utils/fetch-server-config.js';
  * app.use('/auth', delegatedRouter);
  * ```
  *
- * @param metadata The metadata of the authorization server in camelCase format.
+ * @param serverMetadata The metadata of the authorization server in camelCase format.
  * @returns An Express router that serves the metadata at the {@link serverMetadataPaths.oauth}
  * endpoint.
  */
-export const createDelegatedRouter = (metadata: CamelCaseAuthorizationServerMetadata): Router => {
+export const createDelegatedRouter = ({
```

Comment:
will it be better to create a separate function for protected resource routes and mark this func as deprecated?